File: python/src/calisthenics_exercises/one_indentation_level.py
from dataclasses import dataclass


@dataclass
class Order:
    items: list[dict]
    customer_type: str


def process_orders(orders: list[Order]) -> float:
    total_revenue = 0.0
    for order in orders:
        for item in order.items:
            price = item['price']
            quantity = item['quantity']
            subtotal = price * quantity
            if order.customer_type == 'VIP':
                discount = subtotal * 0.1
                subtotal = subtotal - discount
            total_revenue += subtotal
    return total_revenue

File: typescript/src/calisthenics-exercises/one-indentation-level.ts
export function processOrdersWithDiscounts(
  orders: Array<{ id: string; items: Array<{ price: number }>; customer: { isVip: boolean } }>,
): number {
  let total = 0
  for (const order of orders) {
    if (order.items && order.items.length > 0) {
      for (const item of order.items) {
        if (order.customer && order.customer.isVip) {
          if (item.price > 100) {
            total += item.price * 0.8 // gran descuento VIP
          } else {
            total += item.price * 0.9 // pequeño descuento VIP
          }
        } else {
          if (item.price > 100) {
            total += item.price * 0.95 // gran descuento regular
          } else {
            total += item.price // sin descuento
          }
        }
      }
    }
  }
  return total
}
